auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -671,7 +708,166 @@ private MethodSpec createReadMethod(ClassName className,
     return readMethod.build();
   }
 
-  /** Returns a default value for initializing well-known types, or else {@code null}. */
+  private MethodSpec createReadMethod(ClassName className,
+      TypeName autoValueClassName,
+      List<Property> properties,
+      ImmutableMap<TypeName, FieldSpec> adapters,
+      ClassName jsonAdapter,
+      List<TypeVariableName> typeParams,
+      BuilderContext builderContext,
+      ProcessingEnvironment processingEnvironment) {
+    ParameterSpec jsonReader = ParameterSpec.builder(JsonReader.class, "jsonReader").build();
+    MethodSpec.Builder readMethod = MethodSpec.methodBuilder("read")
+        .addAnnotation(Override.class)
+        .addAnnotation(AnnotationSpec.builder(SuppressWarnings.class)
+            .addMember("value", "\"unchecked\"")
+            .build())
+        .addModifiers(PUBLIC)
+        .returns(autoValueClassName)
+        .addParameter(jsonReader)
+        .addException(IOException.class);
+
+    ClassName token = ClassName.get(JsonToken.NULL.getDeclaringClass());
+
+    readMethod.beginControlFlow("if ($N.peek() == $T.NULL)", jsonReader, token);
+    readMethod.addStatement("$N.nextNull()", jsonReader);
+    readMethod.addStatement("return null");
+    readMethod.endControlFlow();
+
+
+    // Instantiate the AutoValue'd class builder
+    FieldSpec builder = FieldSpec.builder(ClassName.get(builderContext.builderType()), "builder")
+        .build();
+
+    Set<ExecutableElement> builderMethods = builderContext.builderMethods();
+
+    if (builderMethods.size() == 0) {
+      // If no builder method defined, instantiate directly.
+      readMethod.addStatement("$T $N = new $T.$L()", builder.type, builder, className,
+          builderContext.builderType().getSimpleName());
+    } else {
+      ExecutableElement builderMethod;
+      if (builderMethods.size() == 1) {
+        // If there is only 1, use it.
+        builderMethod = builderMethods.stream().findFirst().get();
+      } else {
+        // Otherwise, find the only builder method that is annotated.
+        Set<ExecutableElement> annotatedMethods = builderMethods.stream()
+            .filter(e -> MoreElements.isAnnotationPresent(e, AutoValueGsonBuilder.class))
+            .collect(Collectors.toSet());
+
+        if (annotatedMethods.size() == 0) {
+          processingEnvironment.getMessager()
+              .printMessage(
+                  Diagnostic.Kind.ERROR,
+                  "Too many builder methods in " + className.simpleName() + ". Annotate the "
+                      + "builder this TypeAdapter should use with @AutoValueGsonBuilder."
+              );
+          throw new IllegalStateException();
```

Comment:
We should just try to return early and skip generation in this case. Logging the error will defer to the compiler to report issues (it will still fail the build, but with _all_ relevant errors)

Reply:
Same with the below case